src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -223,10 +228,13 @@ private void ThrowIfDisposed()
             // record continuation token for next chunk retrieval
             containerScanInfo.ContinuationToken = blobSegment.ContinuationToken;
 
-            // if ending a cycle then copy currentSweepCycleStartTime to lastSweepCycleStartTime
-            if (blobSegment.ContinuationToken == null)
+            // if ending a cycle then copy currentSweepCycleStartTime to lastSweepCycleStartTime, if changed
+            if (blobSegment.ContinuationToken == null &&
+                containerScanInfo.CurrentSweepCycleLatestModified > containerScanInfo.LastSweepCycleLatestModified)
             {
                 containerScanInfo.LastSweepCycleLatestModified = containerScanInfo.CurrentSweepCycleLatestModified;
+                await _blobScanInfoManager.UpdateLatestScanAsync(container.ServiceClient.Credentials.AccountName,
+                    container.Name, containerScanInfo.LastSweepCycleLatestModified);
```

Comment:
This is currently serializing the last scan timestamp immediately after we've determined the 'new blobs' but before we've executed any functions. With a large number of blobs, this means we may update this, then spend several minutes running functions. If something crashes, we'd skip re-running all of those other blobs and the log scanning would have to pick them up.

It may make more sense to do this after we're done looping through and running functions against the 'new blobs'. Then any restart would just pick back up from the beginning.

I can't think of a downside to that approach but I'll see what others say.